cmake/modules/version.cmake

```diff
@@ -66,6 +66,10 @@ foreach(type file IN ZIP_LISTS VERSION_TYPE VERSION_FILE)
   string(REGEX MATCH "EXTRAVERSION = ([a-z0-9]*)" _ ${ver})
   set(${type}_VERSION_EXTRA ${CMAKE_MATCH_1})
 
+  if(VERSION_TWEAK_OVERRIDE)
+    set(${type}_VERSION_TWEAK ${VERSION_TWEAK_OVERRIDE})
+  endif()
```

Comment:
Wait: why not add the same feature for the kernel too? Even if it ends up rarely ever used, simplicity and symmetry should globally results in LESS code easier to read.

Reply:
It will run for any user of `version.cmake` that sets the define, that APP is the only one to currently do so is not relevant to this code.

The option is of less use for the kernel IMO, whereas the application version is automatically used to sign MCUboot applications